From 0e54e6f5aa11785d76699d8c364105a5d1115c4d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:08:08 -0400
Subject: [powerpc] powernv: Remove unnecessary assignment

Message-id: <1410545655-205645-260-git-send-email-dzickus@redhat.com>
Patchwork-id: 94310
O-Subject: [RHEL7 PATCH 259/626] powerpc/powernv: Remove unnecessary assignment
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 8184616f6fcfc98f0ebf083cbf6a43e5efe54e8a
Author: Gavin Shan <shangw@linux.vnet.ibm.com>
Date:   Thu Dec 26 09:29:40 2013 +0800

    powerpc/powernv: Remove unnecessary assignment

    We don't have IO ports on PHB3 and the assignment of variable
    "iomap_off" on PHB3 is meaningless. The patch just removes the
    unnecessary assignment to the variable. The code change should
    have been part of commit c35d2a8c ("powerpc/powernv: Needn't IO
    segment map for PHB3").

    Signed-off-by: Gavin Shan <shangw@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/pci-ioda.c b/arch/powerpc/platforms/powernv/pci-ioda.c
index d119872..c1b2780 100644
--- a/arch/powerpc/platforms/powernv/pci-ioda.c
+++ b/arch/powerpc/platforms/powernv/pci-ioda.c
@@ -1118,7 +1118,7 @@ void __init pnv_pci_init_ioda_phb(struct device_node *np,
 {
  struct pci_controller *hose;
  struct pnv_phb *phb;
- unsigned long size, m32map_off, iomap_off, pemap_off;
+ unsigned long size, m32map_off, pemap_off, iomap_off = 0;
  const __be64 *prop64;
  const __be32 *prop32;
  int len;
@@ -1205,7 +1205,6 @@ void __init pnv_pci_init_ioda_phb(struct device_node *np,
  size = _ALIGN_UP(phb->ioda.total_pe / 8, sizeof(unsigned long));
  m32map_off = size;
  size += phb->ioda.total_pe * sizeof(phb->ioda.m32_segmap[0]);
- iomap_off = size;
  if (phb->type == PNV_PHB_IODA1) {
   iomap_off = size;
   size += phb->ioda.total_pe * sizeof(phb->ioda.io_segmap[0]);
-- 
1.7.1